pkg/sync/sync_context.go

```diff
@@ -564,9 +564,27 @@ func (sc *syncContext) getSyncTasks() (_ syncTasks, successful bool) {
 
 	sort.Sort(tasks)
 
+	if sc.createNamespace && sc.namespace != "" {
```

Comment:
I don't think this is required. I've checked manually and pre-sync state Namespace is executed first already.